opentelemetry-sdk/src/opentelemetry/sdk/trace/propagation/jaeger_format.py

```diff
@@ -0,0 +1,129 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import typing
+import urllib.parse
+
+import opentelemetry.trace as trace
+from opentelemetry import baggage
+from opentelemetry.context import Context, get_current
+from opentelemetry.trace.propagation.textmap import (
+    Getter,
+    Setter,
+    TextMapPropagator,
+    TextMapPropagatorT,
+)
+
+
+class JaegerFormat(TextMapPropagator):
+    """Propagator for the Jaeger format.
+
+    See: https://www.jaegertracing.io/docs/1.19/client-libraries/#propagation-format
+    """
+
+    TRACE_ID_KEY = "uber-trace-id"
+    BAGGAGE_PREFIX = "uberctx-"
+
+    def extract(
+        self,
+        getter: Getter[TextMapPropagatorT],
+        carrier: TextMapPropagatorT,
+        context: typing.Optional[Context] = None,
+    ) -> Context:
+
+        if context is None:
+            context = get_current()
+        fields = _extract_first_element(
+            getter.get(carrier, self.TRACE_ID_KEY)
+        ).split(":")
+        if len(fields) == 4:
+            trace_id, span_id, _parent_id, flags = fields
+        else:
+            return trace.set_span_in_context(trace.INVALID_SPAN)
+        span = trace.DefaultSpan(
+            trace.SpanContext(
+                trace_id=int(trace_id, 16),
+                span_id=int(span_id, 16),
+                is_remote=True,
+                trace_flags=trace.TraceFlags(
+                    int(flags) & trace.TraceFlags.SAMPLED
+                ),
+            )
+        )
+        baggage_keys = [
+            key
+            for key in getter.keys(carrier)
+            if key.startswith(self.BAGGAGE_PREFIX)
+        ]
+        for key in baggage_keys:
+            value = _extract_first_element(getter.get(carrier, key))
+            context = baggage.set_baggage(
+                key.replace(self.BAGGAGE_PREFIX, ""),
+                urllib.parse.unquote_plus(value).strip(),
+                context=context,
+            )
+        return trace.set_span_in_context(span, context)
+
+    def inject(
+        self,
+        set_in_carrier: Setter[TextMapPropagatorT],
+        carrier: TextMapPropagatorT,
+        context: typing.Optional[Context] = None,
+    ) -> None:
+        span = trace.get_current_span(context=context)
+        span_context = span.get_span_context()
+        if span_context == trace.INVALID_SPAN_CONTEXT:
+            return
+
+        span_parent_id = span.parent.span_id if span.parent else 0
+        trace_flags = span_context.trace_flags
+        # set debug flag to True if sampled flag is set
+        if trace_flags.sampled:
+            trace_flags |= 0x02
```

Comment:
should we support extracting some of the other fields too?

Looking through the SDK today, I don't think there is actually a mechanism where one could implement complete jaeger compatibility. One requirement would be at least propagating trace flags.

Reply:
here's a tracking ticket on the spec side: https://github.com/open-telemetry/opentelemetry-specification/issues/1226. I wouldn't let it block this PR though.